app/src/main/java/eu/kanade/tachiyomi/source/online/english/Mangadex.kt

```diff
@@ -324,6 +326,101 @@ open class Mangadex(override val lang: String, private val internalLang: String,
         return manga
     }
 
+    fun fetchFollows(filters: FilterList) = fetchFollows(0, filters)
+
+    private fun fetchFollows(page: Int, filters: FilterList): Observable<List<Pair<SManga, Int>>> {
+        return getFollowsClient(filters).newCall(followsListRequest(page, filters))
+                .asObservable()
+                .map { response ->
+                    followsParse(response)
+                }.flatMap { parsedResponse ->
+                    // TODO I have no idea if this will work properly or if there's a better way
+                    val currentObservable = Observable.just(parsedResponse.first)
+                    if (parsedResponse.second)
+                        currentObservable.mergeWith(fetchFollows(page + 1, filters))
+                    else
+                        currentObservable
+                }
+    }
+
+    fun followsParse(response: Response): Pair<List<Pair<SManga, Int>>, Boolean> { // TODO exit generic hell
+        val document = response.asJsoup()
+
+        val follows = document.select(followSelector()).map { element ->
+            followFromElement(element)
+        }
+
+        val hasNextPage = followsNextPageSelector()?.let { selector ->
+            document.select(selector).first()
+        } != null
+
+        return Pair(follows, hasNextPage)
+    }
+
+    private fun getFollowsClient(filters: FilterList): OkHttpClient { // TODO reuse/rename `getSearchClient()`?
```

Comment:
currently no difference between this and getSearchClient.  I would just call that for now

Reply:
What about renaming it to something more general

Reply:
Go aheqd